cors.go

```diff
@@ -79,7 +184,17 @@ func AllowedHeaders(headers []string) CORSOption {
 // AllowedMethods ...
 func AllowedMethods(methods []string) CORSOption {
 	return func(ch *cors) error {
-		ch.allowedMethods = methods
+		for _, v := range methods {
+			normalizedMethod := strings.ToUpper(strings.TrimSpace(v))
+			if normalizedMethod == "" {
+				continue
+			}
+
+			if !ch.isMatch(normalizedMethod, ch.allowedMethods) {
+				ch.allowedHeaders = append(ch.allowedHeaders, normalizedMethod)
```

Comment:
I couldn't see anything in the spec that says we have to support it only that a request qualifies as a simple request if its method matches.  So I'm committing a change that makes AllowedMethods a replacement instead of an append operation.